lib/python/rose/apps/rose_ana.py

```diff
@@ -48,6 +50,85 @@
 USRCOMPARISON_EXT = ".py"
 
 
+class KGODatabase(object):
+    """
+    KGO Database object, stores comparison information for rose_ana apps.
+
+    """
+    def __init__(self):
+        """Initialise the object"""
+        self.file_name = os.path.join(
+            os.getenv("ROSE_SUITE_DIR"), "log", "rose-ana-comparisons.db")
+        self.conn = None
+        self.create()
+
+    def get_conn(self):
+        """Return a connection to the database if it exists"""
+        if self.conn is None:
+            self.conn = sqlite3.connect(self.file_name, timeout=60.0)
+        return self.conn
+
+    def create(self):
+        """If the databaste doesn't exist, create it"""
+        self.wait_for_lock()
+        if not os.path.exists(self.file_name):
```

Comment:
Is this logic the right way around? Surely if the database already exists there's no need to lock and immediately unlock it?